sys/include/net/gnrc/ipv6.h

```diff
@@ -11,10 +11,63 @@
  * @ingroup     net_gnrc
  * @brief       GNRC's IPv6 implementation
  *
- * The IPv6 control thread understands messages of type
- *
- *  * @ref GNRC_NETAPI_MSG_TYPE_RCV, and
- *  * @ref GNRC_NETAPI_MSG_TYPE_SND,
+ * This module is for usage with the @ref net_gnrc_netapi
+ *
+ * # NETAPI command documentation
+ *
+ * ## `GNRC_NETAPI_MSG_TYPE_RCV`
+ *
+ * @ref GNRC_NETAPI_MSG_TYPE_RCV expects a @ref net_gnrc_pkt (referred to as
+ * "packet" in the following) in receive order (payload first, headers in
+ * decreasing order by layer). It should at least contain a link-layer payload.
+ * If it contains a @ref net_gnrc_netif_hdr it will be taken into account.
+ *
+ * If the link-layer payload is not an IPv6 packet (starting with a 0x6
+ * half-byte) or the module is not able to parse it correctly it will be
+ * dropped. If the packet can't be forwarded (either to another node or an
+ * application) it will be dropped. Otherwise, there are two scenarios of
+ * success:
+ *
+ * 1. If the destination address is an address on this host, and a suitable
+ *    application is registered to the next header field of the IPv6 header,
+ *    the next header field of one of the IPv6 extension headers within the
+ *    packet or @ref gnrc_nettype_from_protnum() resolves one of the previously
+ *    mentioned next header fields to a @ref net_gnrc_nettype, the packet will
+ *    be forwarded to that application, with the IPv6 header and every IPv6
+ *    extension header between the IPv6 header and the payload marked as
+ *    separate @ref gnrc_pktsnip_t (i.e. the IPv6 payload will be the first
+ *    snip due to receive order). Note, that IPv6-internal headers (such as
+ *    ICMPv6 and extension headers) are handled within this module.
+ * 2. If the network interface the packet was received over (indicated by the
+ *    gnrc_netif_hdr_t::if_pid if provided or the destination address in the
+ *    IPv6 header) is configured as a forwarding interface, the destination is
+ *    not an address on this host, and a route is configured for the destination
+ *    the packet will be forwarded. This means the module will emit a
+ *    `GNRC_NETAPI_MSG_TYPE_SND` with the packet reversed into send order.
+ *
+ * ## `GNRC_NETAPI_MSG_TYPE_SND`
+ *
+ * @ref GNRC_NETAPI_MSG_TYPE_SND expects a @ref net_gnrc_pkt(referred to as
+ * "packet" in the following) in send order (headers first, payload last). It
+ * should at least contain a snip of type @ref GNRC_NETTYPE_IPV6 as its first or
+ * second snip. For the latter case the first snip must be of type @ref
+ * GNRC_NETTYPE_NETIF.
+ *
+ * If the destination address within the @ref GNRC_NETTYPE_IPV6 snip is an
+ * address on this host, the packet will be reversed and merged as though it
+ * came from a network interface (see previous section).
+ *
+ * Otherwise, if a route (or neighbor cache entry) to the destination address
+ * exists, the IPv6 header will be filled for fields that were not set by upper
```

Comment:
"for fields" -> "with fields". Mention, where the values of these fields come from?

Reply:
Tried to but I think this makes it more complicated than it needs to. Essentially: default values, user config or in case of the source address derived from interface configuration and destination address.